### PR TITLE
[Feature] Button Full Width (block prop)

### PR DESCRIPTION
**Pull request by jesskuo4**
## Summary

Closes https://github.com/primer/brand/issues/350

## List of notable changes:

- Added a `block` prop to the `Button` component
- Fixed the playground options for variants for `Button`

## What should reviewers focus on?

-
-

## Steps to test:

1.
1.
1.

## Supporting resources (related issues, external links, etc):

-
-

## Contributor checklist:

- [ ] All new and existing CI checks pass
- [ ] Tests prove that the feature works and covers both happy and unhappy paths
- [ ] Any drop in coverage, breaking changes or regressions have been documented above
- [ ] New visual snapshots have been generated / updated for any UI changes
- [ ] All developer debugging and non-functional logging has been removed
- [ ] Related issues have been referenced in the PR description

## Reviewer checklist:

- [ ] Check that pull request and proposed changes adhere to our [contribution guidelines](../../CONTRIBUTING.md) and [code of conduct](../../CODE_OF_CONDUCT.md)
- [ ] Check that tests prove the feature works and covers both happy and unhappy paths
- [ ] Check that there aren't other open Pull Requests for the same update/change

## Screenshots:

> Please try to provide before and after screenshots or videos

<table>
<tr>
<th> Before</th> <th> After </th>
</tr>
<tr>
<td valign="top">

 </td>
<td valign="top">

</td>
</tr>
</table>

### CODE REVIEW

**Comment by jesskuo4** [comment]
@rezrah Thank you so much for this comment! I'm so sorry I must have missed this, but this is super cool! Let me get to it right now, I haven't tried yet but everything looks manageable and something that I can give a shot. Thank you so much! This will be fun 😎

File: apps/docs/content/components/Button/react.mdx [hunk 1]
```diff
@@ -54,6 +54,14 @@ The `subtle` variant of Button indicates that the action is a low priority one.
 </Stack>
 ```
 
+### Block
+
+```jsx live
+<Stack direction="vertical" alignItems="flex-start">
+  <Button block="true">Block Button</Button>
```
Comment (rezrah) [inline apps/docs/content/components/Button/react.mdx:61]:
```suggestion
  <Button block>
```

File: packages/react/src/Button/Button.features.stories.tsx [hunk 2]
```diff
@@ -84,6 +84,12 @@ export const Large = () => (
   </Button>
 )
 
+export const Block = () => (
+  <Button as="a" variant="primary" block="true" href="#">
```
Comment (rezrah) [inline packages/react/src/Button/Button.features.stories.tsx:88]:
```suggestion
  <Button variant="primary" block>
```

File: package.json [hunk 3]
```diff
@@ -76,5 +76,8 @@
   "engines": {
     "node": ">=16.0.0",
     "npm": ">=8.0.0"
+  },
+  "dependencies": {
+    "@primer/react-brand": "^0.20.1"
```
Comment (rezrah) [inline package.json:81]:
```suggestion
  }
```

File: apps/docs/content/components/Button/index.mdx [hunk 4]
```diff
@@ -73,6 +73,12 @@ Subtle buttons have a transparent background with translucent hover and active s
 
 ![An image showing all subtle states.](https://github.com/primer/brand/assets/912236/7420969a-e561-4fd6-9bef-7340e2b0a7e8)
 
+#### Block
+
+Turning on the block option for button allows the button to span the full width of its container. This feature can be particularly useful when you want the button to occupy more space. Block button can continue to pair with secondary or primary variants.
+
+![An image showing block button with all subtle states.](https://github.com/primer/brand/assets/912236/7420969a-e561-4fd6-9bef-7340e2b0a7e8)
```
Comment (rezrah) [inline apps/docs/content/components/Button/index.mdx:80]: 👋 @jesskuo4 - just a reminder about this image. It's not showing the same image as subtle example above it. Could you swap out for the block version please?

Screenshot 2023-08-08 at 16 21 15

File: apps/docs/content/components/Button/react.mdx [hunk 5]
```diff
@@ -54,6 +54,14 @@ The `subtle` variant of Button indicates that the action is a low priority one.
 </Stack>
 ```
 
+### Block
+
+```jsx live
+<Stack direction="vertical" alignItems="flex-start">
+  <Button block>
+</Stack>
```
Comment (rezrah) [inline apps/docs/content/components/Button/react.mdx:62]:
```suggestion
<Button block>Full-width button</Button>
```

File: apps/docs/content/components/Button/index.mdx (hunk 4 above)
Comment (jesskuo4) [reply apps/docs/content/components/Button/index.mdx:80]: sounds good! thank you @rezrah

**Review by rezrah — APPROVED** [review]
Everything looks great, thank you @jesskuo4 ✨  

Let's ship this... 🚀

**Pull request merged**